aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,576 @@
+and for requests not going down the branch.
+
+Loading middleware options using the options pattern
+----------------------------------------------------
+
+Some modules and handlers have configuration options, which would be stored in the web.config.
+However, in ASP.NET 5 a new configuration model is used in place of web.config.
+
+The new ASP.NET 5 :doc:`configuration system <../fundamentals/configuration>` gives you these options to solve this:
+
+* Use the :ref:`options pattern <options-config-objects>`, as used with MVC. This section shows how.
+* Directly inject the options into the middleware, as shown in the `next section <#loading-middleware-options-through-direct-injection>`_.
+
+1. Create a class to hold your middleware options
+
+    For example:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/MyMiddlewareWithParams.cs
+        :language: c#
+        :linenos:
+        :lines: 8-12
+        :dedent: 4
+
+2. Store the option values
+
+    The new configuration system allows you to essentially store option values anywhere you want. However, most sites will use
+    *appsettings.json*, so lets use that here too:
+
+    .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/appsettings.json
+        :language: json
+        :linenos:
+        :lines: 1, 6-10
+
+    *MyMiddlewareOptionsSection* here is simply a section name. It doesn't have to be the same as the name of your options class.
+
+3. Associate the option values with the options class
+
+    The options pattern uses ASP.NET 5's dependency injection framework to associate the options type (such as ``MyMiddlewareOptions``)
+    with an ``MyMiddlewareOptions`` object that has the actual options.
+
+    To make this work, update your ``Startup`` class:
+
+        a. If you're using *appsettings.json*, make sure it is added to the configuration builder in the ``Startup`` constructor:
+
+        .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+            :language: c#
+            :linenos:
+            :lines: 14-23, 106, 109
+            :emphasize-lines: 7
+            :dedent: 4
+
+        b. Make sure that the options service is configured:
+
+        .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+            :language: c#
+            :linenos:
+            :lines: 14-15, 28-29, 31-33, 43, 109
+            :emphasize-lines: 5
+            :dedent: 4
+
+        c. Associate your options with your options class:
+
+        .. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+            :language: c#
+            :linenos:
+            :lines: 14-15, 28-29, 31-32, 36-39, 43, 109
+            :emphasize-lines: 7,8
+            :dedent: 4
+
+4. Inject the options into your middleware constructor
+
+    Finally inject the options into your middleware's constructor. This is similar to injecting options into a controller.
```

Comment:
No need to indent. Keep this on the same line. Delete finally.

4. Inject the options into your middleware constructor. (not middleware's)